### ISSUE TITLE
XML Registry - error of `limittype` in `VK_NV_fragment_shading_rate_enums` extension

### ISSUE THREAD

**Issue by Pernet** [KhronosGroup/Vulkan-Docs#1767]
I think `limittype` of `maxFragmentShadingRateInvocationCount` in `VkPhysicalDeviceFragmentShadingRateEnumsPropertiesNV` should be `max` (or `noauto`) and not `bitmask`.

Alternatively, if this value is actually a bitmask for valid values (and not just a bit to specify the actual max of valid values), the API should be fixed:
```XML
<member limittype="bitmask"><type>VkSampleCountFlags</type> <name>maxFragmentShadingRateInvocationCount</name></member>
```
instead of
```XML
<member limittype="bitmask"><type>VkSampleCountFlagBits</type> <name>maxFragmentShadingRateInvocationCount</name></member>
```
I can't check the API, since I don't have a GPU with `VK_NV_fragment_shading_rate_enums` extension.

**Comment by Pernet** [comment]
The same type of information seems to be used in `VK_KHR_fragment_shading_rate`.
`maxFragmentShadingRateRasterizationSamples` property in `VkPhysicalDeviceFragmentShadingRatePropertiesKHR` structure is of type `VkSampleCountFlagBits`.
`limittype` attribute of this property is `max`.

I think this is semantically incorrect, but it works in C because `VkSampleCountFlagBits` is an enum whose value in C representation is equal to the sample count:
- `VK_SAMPLE_COUNT_1_BIT` = 0b1 = 1
- `VK_SAMPLE_COUNT_2_BIT` = 0b10 = 2
- `VK_SAMPLE_COUNT_4_BIT` = 0b100 = 4
- `VK_SAMPLE_COUNT_8_BIT` = 0b1000 = 8
etc.

I think all uses of `VkSampleCountFlagBits` should be replaced by an equivalent enum (semantically different, but with the same values):
- `VkImageCreateInfo` struct
- `VkPipelineMultisampleStateCreateInfo` struct
- `VkAttachmentDescription` struct
- `VkPhysicalDeviceSparseImageFormatInfo2` struct
- `VkSampleLocationsInfoEXT` struct
- `VkAttachmentDescription2` struct
- `VkFramebufferMixedSamplesCombinationNV` struct
- `VkPhysicalDeviceFragmentShadingRatePropertiesKHR` struct
- `VkPhysicalDeviceFragmentShadingRateEnumsPropertiesNV` struct
- `VkCommandBufferInheritanceRenderingInfo` struct
- `VkAttachmentSampleCountInfoAMD` struct
- `vkGetPhysicalDeviceSparseImageFormatProperties` command
- `vkGetPhysicalDeviceMultisamplePropertiesEXT` command

I suppose this is a too big change for something that works (in C), so `limittype="max"` should be used for `maxFragmentShadingRateInvocationCount` in `VkPhysicalDeviceFragmentShadingRateEnumsPropertiesNV`.

**Comment by oddhack** [comment]
@pdaniell-nv will evaluate the different cases. In general FlagBits is used if we know that only one of the possible enum values will ever be allowed, so we'll have to check each case to see if that's true or not.

**Comment by pdaniell-nv** [comment]
I have a fix pending for maxFragmentShadingRateInvocationCount. I checked all other occurrences of limittype="bitmask" and it appears this is the only one that is wrong.

I also looked into the other comment about the VkSampleCountFlagBits type being used in structs like VkImageCreateInfo and I think it is correct. VkSampleCountFlagBits is an enum so we want the C API to allow just one of the enum values to be used in places like VkImageCreateInfo. Compare this with VkSampleCountFlags used in structs like VkImageFormatProperties. In that case we want a bitmask of all the possible sample counts supported. @Pernet Did I miss something about your comment?

**Comment by Pernet** [comment]
Imagine an hardware with these possible values for 
- VK_SAMPLE_COUNT_1_BIT
VK_SAMPLE_COUNT_6_BIT
What should be the value for `maxFragmentShadingRateRasterizationSamples` ?

**Comment by Pernet** [comment]
`limittype=max` works for `maxFragmentShadingRateRasterizationSamples` in `VkPhysicalDeviceFragmentShadingRatePropertiesKHR`.
It is correct in a language (like C) in which `max` function can be applied to an `enum`.
If I enumerate toys of my child, I can't say "this one is the max".

**Comment by oddhack** [comment]
This should be fixed in the 1.3.207 spec update.
